quota/etcd/quotaapi/conversions_test.go

```diff
@@ -0,0 +1,215 @@
+		},
+		{
+			desc: "timeBased",
+			mask: timeBasedMask,
+		},
+		{
+			desc:    "sequencingAndTime",
+			mask:    &field_mask.FieldMask{Paths: []string{sequencingBasedPath, timeBasedPath}},
+			wantErr: true,
+		},
+		{
+			desc:    "timeAndSequencing",
+			mask:    &field_mask.FieldMask{Paths: []string{timeBasedPath, sequencingBasedPath}},
+			wantErr: true,
+		},
+		{
+			desc:    "unknownPaths",
+			mask:    &field_mask.FieldMask{Paths: []string{statePath, "NOT_A_FIELD", maxTokensPath}},
+			wantErr: true,
+		},
+		{
+			desc:    "namePath", // readonly
+			mask:    &field_mask.FieldMask{Paths: []string{"name"}},
+			wantErr: true,
+		},
+	}
+	for _, test := range tests {
+		err := validateMask(test.mask)
+		if gotErr := err != nil; gotErr != test.wantErr {
+			t.Errorf("%v: validateMask() returned err = %v, wantErr = %v", test.desc, err, test.wantErr)
+		}
+	}
+}
+
+func TestApplyMask(t *testing.T) {
+	srcSequencingConfig := *apiSequencingConfig
+	destSequencingConfig := *storageTimeConfig // mismatched to test overwrite
+	destSequencingConfig.Name = srcSequencingConfig.Name
+
+	srcTimeConfig := *apiTimeConfig
+	destTimeConfig := *storageSequencingConfig // mismatched to test overwrite
+	destTimeConfig.Name = srcTimeConfig.Name
+
+	destClearSequencing := *storageSequencingConfig
+	wantClearSequencing := destClearSequencing
+	wantClearSequencing.ReplenishmentStrategy = nil
+
+	destClearTime := *storageTimeConfig
+	wantClearTime := destClearTime
+	wantClearTime.ReplenishmentStrategy = nil
+
+	tests := []struct {
+		desc       string
+		src        *quotapb.Config
+		dest, want *storagepb.Config
+		mask       *field_mask.FieldMask
+	}{
+		{
+			desc: "applyToBlank",
+			src:  apiTimeConfig,
+			dest: &storagepb.Config{},
+			mask: &field_mask.FieldMask{Paths: []string{statePath, maxTokensPath}},
+			want: &storagepb.Config{
+				State:     storagepb.Config_DISABLED,
+				MaxTokens: apiTimeConfig.MaxTokens,
+			},
+		},
+		{
+			desc: "sequencingBasedOverwrite",
+			src:  &srcSequencingConfig,
+			dest: &destSequencingConfig,
+			mask: sequencingBasedMask,
+			want: storageSequencingConfig,
+		},
+		{
+			desc: "timeBasedOverwrite",
+			src:  &srcTimeConfig,
+			dest: &destTimeConfig,
+			mask: timeBasedMask,
+			want: storageTimeConfig,
+		},
+		{
+			desc: "clearSequencingIfNil",
+			src:  &quotapb.Config{},
+			dest: &destClearSequencing,
+			mask: &field_mask.FieldMask{Paths: []string{sequencingBasedPath}},
+			want: &wantClearSequencing,
+		},
+		{
+			desc: "clearTimeIfNil",
+			src:  &quotapb.Config{},
+			dest: &destClearTime,
+			mask: &field_mask.FieldMask{Paths: []string{timeBasedPath}},
+			want: &wantClearTime,
+		},
+	}
+	for _, test := range tests {
+		applyMask(test.src, test.dest, test.mask)
+		if !proto.Equal(test.dest, test.want) {
+			t.Errorf("%v: post-applyMask() diff (-got +want):\n%v", test.desc, pretty.Compare(test.dest, test.want))
+		}
+	}
+}
+
+func TestConvert_ApiAndStorage(t *testing.T) {
```

Comment:
["the convention in Go is to use MixedCaps or mixedCaps rather than underscores to write multiword names."](https://golang.org/doc/effective_go.html#mixed-caps)
["This applies even when it breaks conventions in other languages"](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps)